File: src/test/groovy/com/github/jengelman/gradle/plugins/shadow/docs/internal/snippets/executer/CompileException.java
package com.github.jengelman.gradle.plugins.shadow.docs.internal.snippets.executer;

public class CompileException extends RuntimeException {

  private static final long serialVersionUID = 0;

  private final int lineNo;

  public CompileException(Throwable cause, int lineNo) {
    super(cause);
    this.lineNo = lineNo;
  }

  public int getLineNo() {
    return lineNo;
  }

}

File: src/test/kotlin/com/github/jengelman/gradle/plugins/shadow/docs/internal/snippets/executer/CompileException.kt
package com.github.jengelman.gradle.plugins.shadow.docs.internal.snippets.executer

class CompileException(
  cause: Throwable?,
  val lineNo: Int,
) : RuntimeException(cause) {
  private companion object {
    private const val serialVersionUID: Long = 0
  }
}
